main.py

```diff
@@ -1,7 +1,7 @@
 from importlib import reload
 import time
 import os
-
```

Comment:
Does this need to be imported?

Reply:
reload is used if you changed the module, and you don't want to restart the kernel. It's useful in developing/debugging. We could remove it once it's in production. But it does not hurt to keep it.

time is essential for progress report

os - I would keep it, as it correctly inserts backslash for Windows and forward slash in Linux/HDFS between the directory and file name. Still, not essential.

Reply:
I think the first three make sense, but looks like `postcodes_uk` has been imported to (but unused I think?)